File: bidiPhpClient/MessageListener.php
<?php
//namespace MessageService;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of MessageListener
 *
 * @author rarora
 */
require_once __DIR__.'/packages/bidiDemo/MessageService.php';

use bidiDemo\MessageServiceProcessor;

class MessageListener extends Thread{
    private $processor;
    private $protocol;
    
    public function __construct($protocol, $messageServiceHandler){
        $this->protocol = $protocol;
        $this->processor = new MessageServiceProcessor($messageServiceHandler);
    }

    public function run() {
        //while (true) {
            try {
                echo "Client Processor ran.<br/>";
                while (processor.process(protocol, protocol) === true) { }
                echo "Client Processor stopped.";
            } catch (TException $tx) {
                print 'TException: '.$tx->getMessage()."\n";
            }
        //}
    }
}

File: bidiSimpleDemo/src/main/java/bidiSimpleDemo/client/MessageListener.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package bidiSimpleDemo.client;

import bidiSimpleDemo.common.MessageService;
import org.apache.thrift.TException;
import org.apache.thrift.protocol.TProtocol;

/**
 *
 * @author rarora
 */
public class MessageListener implements Runnable{
    private final MessageService.Processor processor;
    private final TProtocol protocol;
    
    public MessageListener(TProtocol protocol, MessageService.Iface messageServiceHandler){
        this.protocol = protocol;
        this.processor = new MessageService.Processor(messageServiceHandler);
    }
     
    @Override
    public void run() {
        while (true) {
        try {
            System.out.println("Client Processor ran.");
            while (processor.process(protocol, protocol) == true) { }
            System.out.println("Client Processor stopped.");
        } catch (TException ex) {
            ex.printStackTrace();
        }
      }
    }
    
}
